### ISSUE TITLE
Error Compiling Static Javascript Files

### ISSUE THREAD

**Issue by Hunga1** [kethinov/roosevelt#70]
When trying to start the application. The following error occurs:
```
Error: ENOENT: no such file or directory, lstat '/Users/username/webapp/statics/js/Users/username/webapp/statics/js/jsFile.js'
```

I'm not using the `jsCompilerWhitelist` configuration. It seems like when Roosevelt attempts to compile all static javascript files, it is appending the absolute path to the statics directory twice. This error seemed to be introduced in the latest version `0.6.30`.

**Comment by kethinov** [comment]
@alallier can you look into this?

**Comment by alallier** [comment]
Yeah I can it's related to what I did. @Hunga1 has his fix and I have mine. I will look at making both work.

**Comment by alallier** [comment]
### Problem

So here's the issue at hand, main problem Windows and secondary problem grabbing/manipulating the files based on non-whitelist/whitelist

#### No Whitelist

When we have no whitelist files specified we take all files specified at the jsPath which defaults to `js` and the files are returned with their **absolute paths**, see this [line](https://github.com/kethinov/roosevelt/blob/master/lib/jsCompiler.js#L14).

##### Windows
So our jsPath (in Windows) is  `C:\Users\username\Documents\project\statics\js\` for example, when we get to this [line](https://github.com/kethinov/roosevelt/blob/master/lib/jsCompiler.js#L43)(s) in the jsCompiler we split on `:` (because that is our [delimiter character](https://github.com/kethinov/roosevelt#statics-parameters)) and grab the left side which gives us (`C`). This causes problems on Windows because the path has drive letter colon in it's path (`C:` in our example). So we do the [spiting ](https://github.com/kethinov/roosevelt#statics-parameters) in that section of the of the jsCompiler and lstat that file which ends up being `C:\Users\alallier\Documents\testingCIGitHub\statics\js\C`. That directory doesn't exist, notice the extra C at the end of the path.

##### Unix

In UNIX we try and split on `:` (assuming a js path of `/Users/username/webapp/statics/js` from @Hunga1's example above) no colon exists and we take to the left of the colon that gives us `/Users/username/webapp/statics/js/jsFile.js` then on the [lstat](/Users/username/webapp/statics/js) line we concat the jsPath (`/Users/username/webapp/statics/js`) + the file (`Users/username/webapp/statics/js/jsFile.js`) which gives us `/Users/username/webapp/statics/js/Users/username/webapp/statics/js/jsFile.js` which doesn't exist.

#### Whitelist

When we whitelist files specified we take all files specified in the [`jsCompilerWhitelist`](https://github.com/kethinov/roosevelt#statics-parameters) parameter and the files are returned with their **absolute paths**, see this [line](https://github.com/kethinov/roosevelt/blob/master/lib/jsCompiler.js#L14).

##### Windows

Everything is fine

##### Unix

Everything is fine

### Possible Solution(s)
I purpose a different delimiter character because the colon does not play nice with Roosevelt (although this is a breaking change) or a Windows/UNIX check with process.platform (easiest non-breaking change)

**Comment by alallier** [comment]
For now I will use process platform to fix the issue

**Comment by kethinov** [comment]
My recommendation is switching the delimiter default to semicolons.

We will need a deprecation procedure though. Check for the presence of colons as a delimiter too. If they are present, then use the old logic and show a deprecation warning saying that the old delimiter will be removed in a future version (probably when we release 1.x).

**Comment by alallier** [comment]
The fix was easier than I thought #74 includes no breaking changes
